common/src/main/java/com/revenuecat/purchases/common/caching/DeviceCache.kt

```diff
@@ -270,28 +284,78 @@ open class DeviceCache(
 
     @Synchronized
     fun setOfferingsCacheTimestampToNow() {
-        offeringsCachedObject.updateCacheTimestamp(Date())
+        offeringsCachedObject.updateCacheTimestamp(dateProvider.now)
```

Comment:
A bit unrelated to these PR, but these methods should be using `dateProvider`. Behavior should remain the same.